torch/_inductor/lowering.py

```diff
@@ -1244,16 +1244,18 @@ def quantized_decomposed_quantize_per_tensor_tensor(
     zero_point_loader = zero_point.make_loader()
 
     def inner_fn(idx):
-        input = input_loader(idx)
+        _input = input_loader(idx)
         _scale = scale_loader((0,) if len(scale.get_size()) == 1 else ())
         _zero_point = zero_point_loader((0,) if len(scale.get_size()) == 1 else ())
-        if scale.dtype != torch.float32:
-            _scale = ops.to_dtype(_scale, torch.float32)
+        if input.dtype != scale.dtype:
+            _input = ops.to_dtype(_input, scale.dtype)
```

Comment:
Casting to double seems like a bad idea for qgpu performance, but it seems to be what eager does so I guess this is fine.

Reply:
Talked to @leslie-fang-intel offline. Let's align with eager which uses float instead of double.

Reply:
In eager implementation, the scales will convert to float before computation. Before refactor, we are using double in inductor. Change back to float to align with eager.
https://github.com/pytorch/pytorch/blob/f35fe4eaf1e9fa2e631f6bf1a3eb6e5fbf14183b/torch/ao/quantization/fx/_decomposed.py#L103
https://github.com/pytorch/pytorch/blob/f35fe4eaf1e9fa2e631f6bf1a3eb6e5fbf14183b/torch/ao/quantization/fx/_decomposed.py#L37-L45